assets/src/apps/delivery/layouts/deck/RestartLessonDialog.tsx

```diff
@@ -62,7 +62,7 @@ const RestartLessonDialog: React.FC<RestartLessonDialogProps> = ({ onRestart })
       const newAttemptUrl = `/sections/${sectionSlug}/page/${revisionSlug}/attempt`;
       redirectTo = newAttemptUrl;
     }
-    if (!graded || isPreviewMode) {
+    if (isPreviewMode) {
       window.location.reload();
     } else {
       window.location.href = redirectTo;
```

Comment:
```suggestion
      window.top.location.href = redirectTo;
```

Let's change this to navigate the top-level window instead of the iframe to prevent that bug we talked about.

Reply:
Done